gulpfile.js

```diff
@@ -88,34 +88,35 @@ var paths = {
   },
 
   vendorChallenges: [
-    'public/bower_components/jshint/dist/jshint.js',
-    'public/bower_components/chai/chai.js',
-    'public/bower_components/CodeMirror/lib/codemirror.js',
-    'public/bower_components/CodeMirror/addon/comment/comment.js',
-    'public/bower_components/CodeMirror/addon/edit/closebrackets.js',
-    'public/bower_components/CodeMirror/addon/edit/matchbrackets.js',
-    'public/bower_components/CodeMirror/addon/lint/lint.js',
-    'public/bower_components/CodeMirror/addon/lint/javascript-lint.js',
-    'public/bower_components/CodeMirror/mode/javascript/javascript.js',
-    'public/bower_components/CodeMirror/mode/xml/xml.js',
-    'public/bower_components/CodeMirror/mode/css/css.js',
-    'public/bower_components/CodeMirror/mode/htmlmixed/htmlmixed.js',
+    'node_modules/jshint/dist/jshint.js',
+    'node_modules/chai/chai.js',
+    'node_modules/codemirror/lib/codemirror.js',
+    'node_modules/codemirror/addon/comment/comment.js',
+    'node_modules/codemirror/addon/edit/closebrackets.js',
+    'node_modules/codemirror/addon/edit/matchbrackets.js',
+    'node_modules/codemirror/addon/lint/lint.js',
+    'node_modules/codemirror/addon/lint/javascript-lint.js',
+    'node_modules/codemirror/mode/javascript/javascript.js',
+    'node_modules/codemirror/mode/xml/xml.js',
+    'node_modules/codemirror/mode/css/css.js',
+    'node_modules/codemirror/mode/htmlmixed/htmlmixed.js',
```

Comment:
@BerkeleyTrue Hm, maybe I'm missing something but I checked and neither `path` nor `require.resolve()` have functionality to solve for non-determinism that you've mentioned. Some observations:

- `require.resolve()` can be effective when calling from code, but when called from top scope, it just returns the global node_modules path for the module if it finds one.
- `path.resolve()` can check if a path exists in multiple sub-directories, but we don't know where npm decides to install the exact version we need, so it won't help.

I don't think there is a way to go around non-determinism with npm, if not use shrinkwrap. For that matter yarn can help and [I've raised issue to consider migration to yarn](https://github.com/FreeCodeCamp/FreeCodeCamp/issues/11501).

An alternative solution would be using cdns in place of all the `js` files that are being complied into `vendor-challenges.js` and `vendor-main.js`, but that may not be desirable for cdn accessibility reasons you mentioned and also increasing the number of network requests significantly.

I'll let it up to you to make the strategic long-term call on this, but happy to help with a PR if you decide on the approach.

Or, if there is an potential solution that I have not listed, would be happy to weight in on it as well.